src/sentry/static/sentry/app/actionCreators/organizations.jsx

```diff
@@ -70,44 +70,47 @@ export function updateOrganization(org) {
   OrganizationsActions.update(org);
 }
 
-export function fetchOrganizationByMember(memberId, {addOrg, fetchOrgDetails}) {
+export async function fetchOrganizationByMember(memberId, {addOrg, fetchOrgDetails}) {
   const api = new Client();
-  const request = api.requestPromise(`/organizations/?query=member_id:${memberId}`);
+  const data = await api.requestPromise(`/organizations/?query=member_id:${memberId}`);
 
-  request.then(data => {
-    if (data.length) {
-      if (addOrg) {
-        // add org to SwitchOrganization dropdown
-        OrganizationsStore.add(data[0]);
-      }
+  if (!data.length) {
+    return null;
+  }
 
-      if (fetchOrgDetails) {
-        // load SidebarDropdown with org details including `access`
-        fetchOrganizationDetails(data[0].slug, {setActive: true, loadProjects: true});
-      }
-    }
-  });
+  const org = data[0];
+
+  if (addOrg) {
+    // add org to SwitchOrganization dropdown
+    OrganizationsStore.add(org);
+  }
 
-  return request;
+  if (fetchOrgDetails) {
+    // load SidebarDropdown with org details including `access`
+    await fetchOrganizationDetails(org.slug, {setActive: true, loadProjects: true});
```

Comment:
You're saying you think this is more correct now right? But that the function will now spend more time resolving since we are no longer returning while the promise is still unresolved for details, correct?

Reply:
Right

Reply:
Okay. I'm good with that.